client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/TabTables.tsx

```diff
@@ -89,21 +90,28 @@ export const QuantityTableComponent: FC<TableProps> = ({
   isDisabled = false,
 }) => {
   const theme = useTheme();
+  const showItemVariantsColumn = useItemVariantsConfigured();
 
   const columns = useColumns<DraftInboundLine>(
     [
       getBatchColumn(updateDraftLine, theme),
       getExpiryColumn(updateDraftLine, theme),
-
-      {
-        key: 'itemVariantId',
-        label: 'label.item-variant',
-        width: 170,
-        Cell: props => (
-          <ItemVariantInputCell {...props} itemId={props.rowData.item.id} />
-        ),
-        setter: updateDraftLine,
-      },
+      ...((showItemVariantsColumn
+        ? [
+            {
+              key: 'itemVariantId',
+              label: 'label.item-variant',
+              width: 170,
+              Cell: props => (
+                <ItemVariantInputCell
+                  {...props}
+                  itemId={props.rowData.item.id}
+                />
+              ),
+              setter: updateDraftLine,
+            },
+          ]
+        : []) as ColumnDescription<DraftInboundLine>[]),
```

Comment:
I slightly prefer the `if (condition) then columns.push(...)` pattern for conditional columns, but that's just minor personal preference. Slightly easier to ready IMHO, but not worth changing.